tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -895,6 +898,203 @@ static void domain_remove_part_context_switch(void)
 	spawn_user();
 }
 
+/*
+ * Stack testing
+ */
+
+#define NUM_STACKS	3
+#define STEST_STACKSIZE	1024
+K_THREAD_STACK_DEFINE(stest_stack, STEST_STACKSIZE);
+K_THREAD_STACK_ARRAY_DEFINE(stest_stack_array, NUM_STACKS, STEST_STACKSIZE);
+
+struct foo {
+	int bar;
+	K_THREAD_STACK_MEMBER(stack, STEST_STACKSIZE);
+	int baz;
+};
+
+struct foo stest_member_stack;
+
+void z_impl_stack_info_get(u32_t *start_addr, u32_t *size)
+{
+	*start_addr = k_current_get()->stack_info.start;
+	*size = k_current_get()->stack_info.size;
+}
+
+Z_SYSCALL_HANDLER(stack_info_get, start_addr, size)
+{
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(start_addr, sizeof(u32_t)));
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(size, sizeof(u32_t)));
+
+	z_impl_stack_info_get((u32_t *)start_addr, (u32_t *)size);
+
+	return 0;
+}
+
+int z_impl_check_perms(void *addr, size_t size, int write)
+{
+	return z_arch_buffer_validate(addr, size, write);
```

Comment:
Could we simply use the Z_SYSCALL_MEMORY_WRITE macros here, too? Isn't it effectively the same?

Reply:
what is the advantage of doing that